newIDE/app/src/VariablesList/index.js

```diff
@@ -244,7 +244,7 @@ export default class VariablesList extends React.Component<Props, State> {
             success = parentVariable.renameChild(name, text);
           }
 
-          this.setState({
+          React.PureComponent({
```

Comment:
Unfortunately this does not make sense.
I would recommend spending more time getting familiar with React, what is a state, what is a prop, how a component works.

Reply:
I don't mind doing that, but why is the tab working then?

Reply:
For now, I think I will close this PR and try to attain a better solution :)

Reply:
It's not really "working", you removed a setState so there was no-rendering of the components, and also no error shown if you enter a bad name :)

Reply:
oh, I think you are right, I will do a course on React and I am sure that  I will get much better at it in a week's time. 
Thanks for the advice.